test/jdk/javax/swing/JFileChooser/CustomApproveButtonTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+import java.awt.Component;
+import java.awt.Container;
+
+import javax.swing.JButton;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+/*
+ * @test
+ * @bug 8299522 8300084
+ * @key headful
+ * @summary Verifies JFileChooser's Approve button text is non-null
+ *          when CUSTOM_DIALOG type is set.
+ * @run main CustomApproveButtonTest
+ */
+
+public class CustomApproveButtonTest {
+    private JFrame frame;
+
+    public static void main(String[] args) throws Exception {
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                UIManager.LookAndFeelInfo[] lookAndFeel = UIManager.getInstalledLookAndFeels();
+                for (UIManager.LookAndFeelInfo look : lookAndFeel) {
+                    new CustomApproveButtonTest(look.getClassName());
+                }
+            }
+        });
+        System.out.println("Test Pass!");
+    }
+
+    private CustomApproveButtonTest(String lookAndFeel) {
+        System.out.println("Testing Look & Feel : " + lookAndFeel);
+
+        setLookAndFeel(lookAndFeel);
+
+        try {
+            frame = new JFrame("CustomApproveButtonTest");
+
+            JFileChooser fileChooser = new JFileChooser();
+            fileChooser.setDialogType(JFileChooser.CUSTOM_DIALOG);
+            frame.add(fileChooser);
+            frame.pack();
+            frame.setVisible(true);
+
+            JButton customApproveButton = fileChooser.getUI().getDefaultButton(fileChooser);
```

Comment:
For Mnemonic the default value is zero, so just have to check if its zero right......?

Reply:
The default value depends on L&F, yet it looks like no L&Fs that are part of JDK set a mnemonic, which makes perfect sense for the default button in a dialog box.

This may call for an additional test which ensures the custom dialog has mnemonic of the Open button.